base/regex.jl

```diff
@@ -364,12 +364,18 @@ function endswith(s::SubString{String}, r::Regex)
 end
 
 function chopprefix(s::AbstractString, prefix::Regex)
+    # fast path to avoid (some, letter-only) regex
+    all(isletter, prefix.pattern) && startswith(s, prefix.pattern) && return chopprefix(s, prefix.pattern)
+
     m = match(prefix, s, firstindex(s), PCRE.ANCHORED)
     m === nothing && return SubString(s)
     return SubString(s, ncodeunits(m.match) + 1)
 end
 
 function chopsuffix(s::AbstractString, suffix::Regex)
+    # fast path to avoid (some) regex, i.e. meant for file endings
+    startswith(raw"\.", prefix.pattern) && all(isletter, prefix.pattern[3:end]) && endswith(s, prefix.pattern[2:end]) && return chopsuffix(s, prefix.pattern)
```

Comment:
- `prefix` undefined
- Indexing with `[2:3]` is unsafe for unicode?
- Indexing with `[2:3]` causes a copy and feels inefficient

Reply:
"prefix" copy-paste error... ran ok, no static analysis... 2 and 3 is ok there in context, because testing if 1 and 2 is an (ASCII) letter.